java-api-core/src/com/kinvey/java/network/NetworkManager.java

```diff
@@ -458,16 +467,20 @@ public String onUnsavedReferenceFound(String collection, GenericJson object) {
 
         if (sourceID != null && !bRealmGeneratedId) {
             save = new Save(entity, myClass, sourceID, SaveMode.PUT);
+            Logger.INFO("Finish for preparing new Save(entity, myClass, sourceID, SaveMode.PUT)");
         } else {
             save = new Save(entity, myClass, SaveMode.POST);
+            Logger.INFO("Finish for preparing new Save(entity, myClass, SaveMode.POST)");
         }
 
         client.initializeRequest(save);
```

Comment:
Let's add a log for just before the request initialization begins.

Reply:
Added